src/lib/server/orders.ts

```diff
@@ -659,6 +659,19 @@ export async function createOrder(
 		throw error(400, "You can't use free payment method on this order");
 	}
 
+	items.forEach((item) => {
+		if (item.chosenVariations && item.product.variations) {
```

Comment:
We don't care if `item.chosenVariations`, we only care if `item.product.variations?.length`.

If the backend added variations between add to cart & order creation, we should still prevent the order from being created if there are no chosen variations.